.github/workflows/test_on_push.yml

```diff
@@ -86,4 +87,36 @@ jobs:
         run: python -m pip install nox
 
       - name: Check if the documentation can be built
-        run: python -m nox -s docs
+        run: nox -s docs
+
+  generated_project_tests:
+      needs: [template_test]
+
+      runs-on: ${{ matrix.os }}
+      strategy:
+        fail-fast: false
+        matrix:
+          os: [ubuntu-latest, macos-13, macos-14, windows-latest]
+          python-version: ["3.9", "3.10", "3.11", "3.12"]
+      name:
+        Generated Project Tests (${{ matrix.os }} / Python ${{ matrix.python-version }})
+      steps:
+        - name: Checkout pybamm-cookiecutter
+          uses: actions/checkout@v4
+
+        - name: Set up Python ${{ matrix.python-version }}
+          uses: actions/setup-python@v5
+          with:
+            python-version: ${{ matrix.python-version }}
+
+        - name: Install cookiecutter and pipx
+          run: python -m pip install cookiecutter pipx
+
+        - name: Generate a project using cookiecutter
+          run: cookiecutter . --no-input
+
+        - name: Install nox and test generated project
+          working-directory: ./pybamm-example-project
+          run: |
+            pipx install nox
+            pipx run nox -s generated-project-tests
```

Comment:
We're using `uv` to install `nox` and the nox CLI to execute tests in the job above. Here we're using `pipx` to do our work. Could you make this uniform, please? (use any of the one package downloader/management tool)

Reply:
+1. Side note: `pipx run nox` means that `pipx install nox` becomes redundant. `pipx run` runs commands from CLI apps without having to install them (though it does install them in a temporary cache, of course, if we are to be pedantic)

Reply:
I was using `pipx` as a temporary fix for failing `generated-project-tests`, the CI was returning `nox command not found` when the working directory was changed, it is probably a bug because I found no trace of a similar issue on nox's github page or any threads related to github-actions. [Here](https://github.com/pybamm-team/pybamm-cookiecutter/actions/runs/10125311771/job/28000668991?pr=29) is the failed CI logs of the most recent commit where I tried executing the nox installation commands after changing the working directory. [Here](https://github.com/pybamm-team/pybamm-cookiecutter/actions/runs/10124950232/job/27999920383) are the logs for the previous CI steps where nox installation and changing the working directory where two different steps in the CI. I am not sure how to proceed with this issue. I tried the `uvx` experimental build to run nox session locally, but that doesn't seem to work in CI as well.

Reply:
Okay, so I got it working by using [nox's github-action](https://github.com/wntrblm/nox/blob/main/action.yml) `wntrblm/nox@2024.04.15` to setup and install nox. It uses `uv` by default to install the packages so it also preserves uniformity in the workflow.